examples/pipelines/fl_cross_silo_factory/submit.py

```diff
@@ -282,132 +318,58 @@ def silo_scatter_subgraph(
         # DP max gradient norm
         dp_max_grad_norm=dp_max_grad_norm,
         # Total number of iterations
-        total_num_of_iterations=total_num_of_iterations,
+        total_num_of_iterations=num_of_iterations,
         # Silo name/identifier
-        metrics_prefix=scatter_name,
+        metrics_prefix=silo_name,
         # Iteration number
         iteration_num=iteration_num,
     )
 
     # Assigning the silo's second compute to the training component
-    silo_training_step.compute = scatter_compute2
+    silo_training_step.compute = silo_compute2
 
-    # IMPORTANT: we will assume that any output provided here can be exfiltrated into the orchestrator
+    # IMPORTANT: we will assume that any output provided here can be exfiltrated into the orchestrator/gather
     return {
         # NOTE: the key you use is custom
         # a map function scatter_to_gather_map needs to be provided
         # to map the name here to the expected input from gather
-        "input_silo": silo_training_step.outputs.model
+        "model": silo_training_step.outputs.model
     }
 
 
-#######################
-### D. FACTORY CODE ###
-#######################
-
-# In this section, we're using a wrapper to build the full FL pipeline
-# based on the custom methods you implemented in section C.
+def get_scatter_constant_inputs():
+    """Get scatter constant inputs like lr, epochs, etc."""
+    return YAML_CONFIG.inputs
 
-# 1. create an instance of the factory
-builder = FederatedLearningPipelineFactory()
 
-# 2. feed it with FL parameters
+#######################
+### D. FL Contract ###
+#######################
 
-builder.set_orchestrator(
-    # provide settings for orchestrator
-    YAML_CONFIG.federated_learning.orchestrator.compute,
-    YAML_CONFIG.federated_learning.orchestrator.datastore,
-)
+from fl_helper import scatter_gather
 
-for silo_config in YAML_CONFIG.federated_learning.silos:
-    builder.add_silo(
-        # provide settings for this silo
-        silo_config.name,
-        silo_config.computes,
-        silo_config.datastore,
-        # any additional custom kwarg will be sent to silo_preprocessing() as is
-        raw_train_data=Input(
-            type=silo_config.training_data.type,
-            mode=silo_config.training_data.mode,
-            path=silo_config.training_data.path,
-        ),
-        raw_test_data=Input(
-            type=silo_config.testing_data.type,
-            mode=silo_config.testing_data.mode,
-            path=silo_config.testing_data.path,
-        ),
-    )
+scatter_configs = get_scatter_configs()
+gather_config = get_gather_config()
```

Comment:
same here, I'd unpack this function here, not wrap up the gather config parsing in a function.